### PR TITLE
feat(awslambda): add new fixer `awslambda_function_not_publicly_accessible_fixer`

### PR DESCRIPTION
**Pull request by danibarranqueroo**
### Context

Created a fixer that restricts public access to Lambda functions, ensuring that they are only accessible through private networks or specific trusted resources. This will help prevent unauthorized access to sensitive data and functionality within the Lambda functions.

### Description

Added new fixer `awslambda_function_not_publicly_accessible_fixer` with its unit tests.

### Checklist

- Are there new checks included in this PR? Yes / No
    - If so, do we need to update permissions for the provider? Please review this carefully.
- [ ] Review if the code is being covered by tests.
- [ ] Review if code is being documented following this specification https://github.com/google/styleguide/blob/gh-pages/pyguide.md#38-comments-and-docstrings
- [ ] Review if backport is needed.

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: prowler/providers/aws/services/awslambda/awslambda_function_not_publicly_accessible/awslambda_function_not_publicly_accessible.metadata.json [hunk 1]
```diff
@@ -3,7 +3,7 @@
   "CheckID": "awslambda_function_not_publicly_accessible",
   "CheckTitle": "Check if Lambda functions have resource-based policy set as Public.",
   "CheckType": [],
-  "ServiceName": "lambda",
+  "ServiceName": "awslambda",
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/awslambda/awslambda_function_not_publicly_accessible/awslambda_function_not_publicly_accessible.metadata.json:6]: Good catch! Check if there are more cases like this one and fix them, please.

File: prowler/providers/aws/services/awslambda/awslambda_function_not_publicly_accessible/awslambda_function_not_publicly_accessible_fixer.py
Comment (MrCloudSec) [inline prowler/providers/aws/services/awslambda/awslambda_function_not_publicly_accessible/awslambda_function_not_publicly_accessible_fixer.py:1]: @danibarranqueroo there are different ways to make a Lambda's policy public, not just the Principal being `*`. For simplicity, I think the best option is to remove the entire Lambda policy.

Comment (danibarranqueroo) [reply prowler/providers/aws/services/awslambda/awslambda_function_not_publicly_accessible/awslambda_function_not_publicly_accessible_fixer.py:1]: Okay, thanks for notifying that, as there is no way to remove the entire policy I've done it removing one by one all the Statements.

**Review by MrCloudSec — CHANGES_REQUESTED** [review]
Please, also add a permission to the function to allow access from the AWS account:

```
aws lambda add-permission
  --region region
  --function-name resource-id
  --statement-id ProwlerFixerStatement
  --principal account-id
  --action lambda:InvokeFunction
```

File: prowler/providers/aws/services/awslambda/awslambda_function_not_publicly_accessible/awslambda_function_not_publicly_accessible_fixer.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        ]
+    }
+    Args:
+        resource_id (str): The Lambda function name or ARN.
+        region (str): AWS region where the Lambda function exists.
+    Returns:
+        bool: True if the operation is successful (policy removed and permission added), False otherwise.
+    """
+    try:
+        account_id = get_account_id()
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/awslambda/awslambda_function_not_publicly_accessible/awslambda_function_not_publicly_accessible_fixer.py]: Please, get the account from `awslambda_client.audited_account`.

File: prowler/providers/aws/services/awslambda/awslambda_function_not_publicly_accessible/awslambda_function_not_publicly_accessible_fixer.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+        regional_client = awslambda_client.regional_clients[region]
+        policy_response = regional_client.get_policy(FunctionName=resource_id)
+        policy = policy_response.get("Policy")
+
+        if policy:
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/awslambda/awslambda_function_not_publicly_accessible/awslambda_function_not_publicly_accessible_fixer.py]:
```suggestion
        policy_response = regional_client.get_policy(FunctionName=resource_id)
        policy = json.loads(
                            policy_response.get("Policy","")
                        )
```

File: prowler/providers/aws/services/awslambda/awslambda_function_not_publicly_accessible/awslambda_function_not_publicly_accessible_fixer.py [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+            if isinstance(policy, str):
+                policy = json.loads(policy)
+
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/awslambda/awslambda_function_not_publicly_accessible/awslambda_function_not_publicly_accessible_fixer.py]:
```suggestion
```

It is always a string.

**Review by MrCloudSec — CHANGES_REQUESTED** [review]
Please, review my comments.

File: prowler/providers/aws/services/awslambda/awslambda_function_not_publicly_accessible/awslambda_function_not_publicly_accessible_fixer.py (hunk 3 above)
Comment (danibarranqueroo) [reply prowler/providers/aws/services/awslambda/awslambda_function_not_publicly_accessible/awslambda_function_not_publicly_accessible_fixer.py]: Done!

File: prowler/providers/aws/services/awslambda/awslambda_function_not_publicly_accessible/awslambda_function_not_publicly_accessible_fixer.py (hunk 2 above)
Comment (danibarranqueroo) [reply prowler/providers/aws/services/awslambda/awslambda_function_not_publicly_accessible/awslambda_function_not_publicly_accessible_fixer.py]: Didn't know that!! Just done it 🔝

**Review by MrCloudSec — APPROVED** [review]

**Pull request merged**